### PR TITLE
Upgrade to Polkadot v1.0.0

### PR DESCRIPTION
**Pull request by jasl**
https://github.com/paritytech/substrate/compare/polkadot-v0.9.43...polkadot-v1.0.0
https://github.com/paritytech/cumulus/compare/polkadot-v0.9.43...polkadot-v1.0.0
https://github.com/paritytech/polkadot/releases/tag/v1.0.0
https://github.com/Phala-Network/phala-blockchain/pull/1363

This is a giant PR

- Sync Phala pallet to https://github.com/Phala-Network/phala-blockchain/commit/d4d4a2571f870ac2fae2aeb99690852100601296
- Use Rust 1.72.0

https://github.com/Phala-Network/phala-blockchain/pull/1391 requires migration

No upstream migration is needed
- https://github.com/paritytech/polkadot/blob/release-v1.0.0/runtime/kusama/src/lib.rs#L1550-L1561
- https://github.com/paritytech/polkadot/blob/release-v1.0.0/runtime/polkadot/src/lib.rs#L1517-L1524
- These are for Gov1
  - https://github.com/paritytech/substrate/pull/14241
  - https://github.com/paritytech/substrate/pull/14226
  - https://github.com/paritytech/substrate/pull/14218
  - https://github.com/paritytech/substrate/pull/14241

### CODE REVIEW

File: pallets/phala-world/src/nft_sale.rs [hunk 1]
```diff
@@ -1572,20 +1573,20 @@ where
...
 			origin_of_shell_collection_id,
-			nft_id,
+			nft_id.into(),
 		)?;
 
 		Self::deposit_event(Event::OriginOfShellMinted {
 			rarity_type,
-			collection_id: origin_of_shell_collection_id,
+			collection_id: origin_of_shell_collection_id.into(),
 			nft_id,
```
Comment (HashWarlock) [inline pallets/phala-world/src/nft_sale.rs:1582]: I see most have `nft_id.into()` but others do not. Do they all need to have `.into()`?

Comment (jasl) [reply pallets/phala-world/src/nft_sale.rs:1582]: Basically PW pallet use `CollectionId` and `NftId` from `rmrk-primitives`, which are not traits type `CollectionId` and `ItemId`, so it has to convert.

I do not change PW codes too much, only add `into()` when the compiler says error, but I may over-done this.

**Review by tolak — APPROVED** [review]
SubBridge & inDEX looks good to me, btw already update sygma to include [this fix](https://github.com/sygmaprotocol/sygma-substrate-pallets/pull/127)

**Comment by jasl** [comment]
@h4x3rotab I'm merging this now, feel free to review later

**Pull request merged**